Add release management docs (#4139)
Napari/napari: docs/developers/release.md

Currently, it only handles distribution, but as the project matures,
it will include generating release notes, documentation, etc.

# Timeline

> New versions of `napari` will be released every two months. The first release candidate will be available one week prior to release for testing purposes. Multiple release candidates may become available during the week prior to release.

Reviewer: Do we want to actually say that the rc will become available first Tuesday of the month or no?

Author: Good point! Will add when on a computer

> New versions of `napari` will be released every two months. The first release candidate will be available one week prior to release for testing purposes. Multiple release candidates may become available during the week prior to release.

Author: ```suggestion
New versions of `napari` will be released every two months. The first release candidate will be available one week prior to release for testing purposes. Multiple release candidates may become available during the week prior to release. Upcoming releases can be found in our public calendar.
```

The latest release candidate can be installed with 

`pip install --pre napari`

# Release management
The release will be coordinated by a release manager whose responsibilities include...

## Two weeks before release (one week before release candidate)
- Look through currently open PRs and get a sense of what would be good to merge before the first release candidate 
- Create a zulip thread in the release channel letting people know the release candidate is coming and pointing out PRs that would be nice to merge before release

> At this stage, bug fixes should and features that are close to landing should be prioritised. The release manager will follow up with PR authors, reviewing and merging as needed.

Reviewer: ```suggestion
At this stage, bug fixes and features that are close to landing should be prioritized. The release manager will follow up with PR authors, reviewing and merging as needed.
```

I think we use `z`s and skip out on `u`s around here 😆 Just kidding but to be consistent we should use US spelling across the board

  
## Nine days before release (two days before release candidate)
- Generate release notes with the script in the release folder
- Fill in the release highlights and make a PR with the release notes

> At this point the release manager should really now be the only person merging PRs on the repo for the next week.

Reviewer: ```suggestion
At this point the release manager should ideally be the only person merging PRs on the repo for the next week.
```

## One week before release 
- Add any recently merged PRs to release notes
- Merge release notes
- Make the release candidate
- Announce to release stream on zulip that the first release candidate is available for testing

## The week before release
- Merge any PRs and update release notes accordingly
- Make new release candidates as necessary and announce them on zulip

> At this stage PRs merged should focus mainly on regressions and bug fixes. new features should wait until after release.

Reviewer: ```suggestion
At this stage PRs merged should focus mainly on regressions and bug fixes. New features should wait until after release.
```
